### PR TITLE
#388 disable editable field to attendee

### PR DESCRIPTION
**Pull request by asmaNode**
**TO DO:** 

- [x] test

**DEMO**

https://user-images.githubusercontent.com/43316837/106864665-7d818800-66ca-11eb-877d-7c790dce3f84.mp4

### CODE REVIEW

File: src/esn.calendar.libs/app/services/cached-event-source.js [hunk 1]
```diff
@@ -182,6 +188,7 @@ require('./cal-ui-authorization-service.js');
     function setEditable(calendar, events) {
       return $q.all(events.map(function(event) {
         return calUIAuthorizationService.canModifyEvent(calendar, event, session.user._id).then(function(canModifyEvent) {
+
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/cached-event-source.js:191]:
```suggestion
```

File: src/esn.calendar.libs/app/services/cached-event-source.js [hunk 2]
```diff
@@ -79,6 +79,12 @@ require('./cal-ui-authorization-service.js');
             change.instances = [];
             change.event.expand(start.clone().subtract(1, 'day'), end.clone().add(1, 'day')).forEach(function(subEvent) {
               if (subEvent.status !== 'CANCELLED') {
+
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/cached-event-source.js:82]:
```suggestion
```

**Comment by rezk2ll** [comment]
LGTM

**Comment by huy-ta** [comment]
I think this should be OK.

File: src/esn.calendar.libs/app/services/cached-event-source.js (hunk 2 above)
Comment (rezk2ll) [reply src/esn.calendar.libs/app/services/cached-event-source.js:82]: this is not resolved, the blank line is unnecessary.

check: https://arcticicestudio.github.io/styleguide-javascript/rules/whitespace.html#padded-blocks

Comment (asmaNode) [reply src/esn.calendar.libs/app/services/cached-event-source.js:82]: it is resolved in my second commit

File: src/esn.calendar.libs/app/services/cached-event-source.spec.js [hunk 3]
```diff
@@ -384,6 +384,29 @@ describe('The calCachedEventSource service', function() {
         expect(self.originalCallback).to.have.been.calledWithExactly(self.events.concat(correctSubEvent));
       });
 
+      it('should set subEvents as non editable if the parent recurrent event is not editable ', function() {
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/cached-event-source.spec.js:387]:
```suggestion
      it('should set subEvents as non editable if the parent recurrent event is not editable', function() {
```

File: src/esn.calendar.libs/app/services/cached-event-source.spec.js [hunk 4]
```diff
@@ -384,6 +384,29 @@ describe('The calCachedEventSource service', function() {
...
+          uid: self.modifiedEvent.id,
+          start: self.start.clone().add(1, 'hour'),
+          calendarUniqueId: self.calendarUniqueId,
+          end: self.end.clone().subtract(1, 'hour'),
+          isInstance: _.constant(true),
+          isRecurring: _.constant(false)
+        };
+
+        self.modifiedEvent.editable = false;
+
```
Comment (rezk2ll) [inline src/esn.calendar.libs/app/services/cached-event-source.spec.js:399]:
```suggestion
```

**Review by huy-ta — APPROVED** [review]

**Review by rezk2ll — APPROVED** [review]

**Pull request merged**